addons/pinniped/post-deploy/pkg/utils/utils_test.go

```diff
@@ -0,0 +1,130 @@
+// Copyright 2021 VMware, Inc. All Rights Reserved.
+// SPDX-License-Identifier: Apache-2.0
+
+package utils
+
+import (
+	"context"
+	"errors"
+	"testing"
+
+	certmanagerv1 "github.com/jetstack/cert-manager/pkg/apis/certmanager/v1"
+	"github.com/stretchr/testify/require"
+	corev1 "k8s.io/api/core/v1"
+	kubeerrors "k8s.io/apimachinery/pkg/api/errors"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/runtime"
+	kubefake "k8s.io/client-go/kubernetes/fake"
+	kubetesting "k8s.io/client-go/testing"
+)
+
+func TestRemoveDefaultTLSPort(t *testing.T) {
```

Comment:
Would it be useful at all to test that we properly bubble up any errors from `url.Parse`?

Reply:
sure yeah i could be down for that...will update this pr with a new commit

Reply:
added efb2398d1294a6410ad71885abe7e323a56fb245 to try to cover this. let me know what you think

Reply:
Looks great! Thank you!